### PR TITLE
ablation for op

### PR DESCRIPTION
**Pull request by Angryrou**
1. drop the th.exp(X) for the MLP output
2. add a skip connection MLP 
3. fix a bug in the LRscheduler 
4. some quick observations from hyperparameter tuning results for the new structures (with `dropout=0.1, lr=1e-3, osize=32`)
   - skip_connection did help than basic MLP
   - `eps=200` and `300` are averagely close, but 300 has a better chance of hitting better results
   - adding capacity `(nlayers, nhidden)` does not necessarily help a lot in the testing range. 
   - with dropout, turning off batchnorm is better for the performance
   - weight decay options does not affect the performance a lot. `wd = 1e-2` tends to hit better performance in the distribution.
    <details><summary>hyper parameter distributions</summary>image</details>

5. at the second round of hyperparameter tuning, I fixed `dropout=0.1, lr=1e-3, eps=300, use_bn=False, wd=1e-2`. Try `osize` in `[16, 32, 64, 128]`, and tried `(nlayers, nhidden)` in `[(4, 256), (4, 512), (5, 256), (5, 512)]`.

### CODE REVIEW

**Comment by Angryrou** [comment]
* **#81** <a href="https://app.graphite.dev/github/pr/Angryrou/udao-spark-optimizer/81?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Angryrou and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>
